### PR TITLE
Add totalview@2020.3.11 for BB5

### PR DESCRIPTION
**Pull request by pramodk**
* add new package version for totalview
 * change mirror handling : rsync whole mirror dir
 * enable totalview module

Tarball has been added to:

```
/gpfs/bbp.cscs.ch/ssd/apps/hpc/download/totalview/
```

### CODE REVIEW

File: deploy/environments/externals.yaml [hunk 1]
```diff
@@ -73,6 +73,7 @@ spack:
     - stat@4.0.2.999
     - symengine@0.3.0
     - timemory
+    - totalview@2020.3.11
```
Comment (matz-e) [inline deploy/environments/externals.yaml]: Shouldn't we remove the version to always grab the latest?

File: var/spack/repos/builtin/packages/totalview/package.py [hunk 2]
```diff
@@ -43,10 +46,10 @@ class Totalview(Package):
     # resource once the target architecture is known.
     resource(
         name='x86_64',
-        url='file://{0}/totalview_2020.3.11_linux_x86-64.tar'.format(os.getcwd()),
+        url='file:///gpfs/bbp.cscs.ch/apps/hpc/download/totalview/totalview_2021.1.16_linux_x86-64.tar',
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/totalview/package.py]:
```suggestion
        url='file://{0}/totalview_2020.3.11_linux_x86-64.tar'.format(mirror_dir),
```

**Review by matz-e — CHANGES_REQUESTED** [review]

**Comment by pramodk** [comment]
Mirror in env might need different handling: https://spack.readthedocs.io/en/latest/mirrors.html#mirror-environment

**Comment by matz-e** [comment]
Some conclusions:
* Package downloads on the mirror need to be of form `{mirrror_url}/{package_name}/{package_name}-{version}.tar`
* Resource downloads on the mirror need to be of form `{mirror_url}/{package_name}/{resource_name}-{version}.tar`

Got stuff to build with a mirror setup where the `totalview` directory looked like:
```
-rw-r--r-- 1 matwolf bbp       950 May  4 14:46 totalview.2020.3.11.md5sum
-rw-r--r-- 1 matwolf bbp 253839360 May  4 14:46 x86_64-2021.1.16.tar
-rw-r--r-- 1 matwolf bbp       950 May  4 14:46 totalview.2021.1.16.md5sum
-rw-r--r-- 1 matwolf bbp  28057600 May  4 14:46 totalview.2020.3.11.tar
-rw-r--r-- 1 matwolf bbp   1001091 May  4 14:46 totalview.2020.3-release_notes.pdf
-rw-r--r-- 1 matwolf bbp 249026560 May  4 14:46 totalview_2020.3.11_linux_x86-64.tar
-rw-r--r-- 1 matwolf bbp  29163520 May  4 14:46 totalview-2021.1.16.tar
-rw-r--r-- 1 matwolf bbp    436714 May  4 14:46 totalview.2021.1-release_notes.pdf
```

**Comment by matz-e** [comment]
Files on the mirror have been adjusted.

Retest this please.

**Review by matz-e — APPROVED** [review]
All good from my side.

**Review by iomaganaris — APPROVED** [review]

**Comment by pramodk** [comment]
@matz-e : didn't realise this was pending. I will merge this now as there are no conflicts. Lets see if something else will fail.

**Pull request merged**